pandas/core/ops.py

```diff
@@ -386,71 +386,102 @@ def __init__(self, left, right, name, na_op):
-              (self.is_timedelta_lhs or self.is_offset_lhs)):
+        elif self.is_datetime_rhs:
             if name not in ('__add__', '__radd__', '__rsub__'):
                 raise TypeError("can only operate on a timedelta/DateOffset "
                                 "with a rhs of a datetime for addition, "
                                 "but the operator [{name}] was passed"
                                 .format(name=name))
+        else:
+            raise TypeError('cannot operate on a series without a rhs '
+                            'of a series/ndarray of type datetime64[ns] '
+                            'or a timedelta')
 
-        # 2 datetimes
-        elif self.is_datetime_lhs and self.is_datetime_rhs:
-
-            if name not in ('__sub__', '__rsub__'):
-                raise TypeError("can only operate on a datetimes for"
-                                " subtraction, but the operator [{name}] was"
-                                " passed".format(name=name))
+    def _validate_offset(self, name):
+        # assumes self.is_offset_lhs
 
-            # if tz's must be equal (same or None)
-            if getattr(lvalues, 'tz', None) != getattr(rvalues, 'tz', None):
-                raise ValueError("Incompatible tz's on datetime subtraction "
-                                 "ops")
-
-        elif ((self.is_timedelta_lhs or self.is_offset_lhs) and
-              self.is_datetime_rhs):
+        if self.is_timedelta_rhs:
+            # 2 timedeltas
+            if name not in ('__div__', '__rdiv__', '__truediv__',
+                            '__rtruediv__', '__add__', '__radd__', '__sub__',
+                            '__rsub__'):
+                raise TypeError("can only operate on a timedeltas for addition"
+                                ", subtraction, and division, but the operator"
+                                " [{name}] was passed".format(name=name))
 
+        elif self.is_datetime_rhs:
             if name not in ('__add__', '__radd__'):
                 raise TypeError("can only operate on a timedelta/DateOffset "
                                 "and a datetime for addition, but the operator"
                                 " [{name}] was passed".format(name=name))
+
+        else:
+            raise TypeError('cannot operate on a series without a rhs '
+                            'of a series/ndarray of type datetime64[ns] '
+                            'or a timedelta')
+
+    def _validate(self, lvalues, rvalues, name):
+        if self.is_datetime_lhs:
+            return self._validate_datetime(lvalues, rvalues, name)
+        elif self.is_timedelta_lhs:
+            return self._validate_timedelta(name)
+        elif self.is_offset_lhs:
+            return self._validate_offset(name)
+
+        if ((self.is_integer_lhs or self.is_floating_lhs) and
+                self.is_timedelta_rhs):
+
+            if name not in ('__div__', '__truediv__', '__mul__', '__rmul__'):
```

Comment:
maybe factor this to a function to avoid duplication with above, 

``assert_can_operate()`` or somesuch